zip-0312.rst

```diff
@@ -1,7 +1,466 @@
+- The zero-knowledge proof for the transaction is created with the randomizer
+  as an auxiliary (secret) input, among others.
+
+When employing re-randomizable FROST as specified in this ZIP, the goal is to
+split the spend authorization private key :math:`\mathsf{ask}` among multiple
+possible signers. This means that the proof generation will still be generated
+by a single participant, likely the one that created the transaction in the first
+place. Note that this user already controls the privacy of the transaction since
+they are responsible for creating the proof.
+
+This fits well into the "Coordinator" role from the FROST specification [#frost-protocol]_.
+The Coordinator is responsible for sending the message to be signed to all participants,
+and to aggregate the signature shares.
+
+With those considerations in mind, the threat model considered in this ZIP is:
+
+- The Coordinator is trusted with the privacy of the transaction (which includes
+  the unlinkability property). A rogue Coordinator will be able to break
+  unlinkability and privacy, but should not be able to create signed transactions
+  without the approval of `MIN_SIGNERS` participants, as specified in FROST.
+- All key share holders are also trusted with the privacy and of the transaction,
+  thus a rogue key share holder will be able to break its privacy and unlinkability.
+  A future specification may support a scenario where individual key share
+  holders are not trusted with it.
+
+
+Non-requirements
+================
+
+- This ZIP does not support removing the Coordinator role, as described in #[frost-removingcoordinator]_.
+- This ZIP does not prevent key share holders from linking the signing operation to a transaction in the blockchain.
+- Like the FROST specification [#FROST], this ZIP does not specify a key generation
+  procedure; but refer to that specification for guidelines.
+- Network privacy is not in scope for this ZIP, and must be obtained with other
+  tools if desired.
+
+
+Specification
+=============
+
+Algorithms in this section are specified using Python pseudo-code, in the same
+fashion as the FROST specification [#FROST]_.
+
+The types Scalar, Element, and G are defined in #[frost-primeordergroup]_, as well
+as the notation for elliptic-curve arithmetic, which uses the additive notation.
+
+
+Re-randomizable FROST
+---------------------
+
+To add re-randomization to FROST, follow the specification [#FROST]_ with the
+following modifications.
+
+
+Key Generation
+''''''''''''''
+
+While key generation is out of scope for this ZIP and the FROST spec [#FROST]_,
+it needs to be consistent with FROST, see [#frost-tdkg]_ for guidance. The
+spend authorization private key :math:`\mathsf{ask}` [#protocol-spendauthsig]_
+is the particular key that must be used in the context of this ZIP. Note that
+while Sapling allows creating it directly, in Orchard it is always derived
+from the spending key :math:`\mathsf{ask}`. This means that a trusted
```

Comment:
$\mathsf{ask}$ is derived from $\mathsf{ask}$? That sounds wrong.

Reply:
It's not a consensus requirement of either Sapling or Orchard that  be generated in any particular way. In practice it is generated according to ZIP 32 in the single-signer case, but that doesn't really constrain how it is generated in the multi-signer case.